src/ray/gcs/gcs_client/service_based_accessor.h

```diff
@@ -9,6 +10,41 @@ namespace gcs {
 
 class ServiceBasedGcsClient;
 
+/// \class Sequencer
+/// Sequencer guarantees that all operations with the same ordering key are sequenced.
+/// This class is thread safe.
+template <class KEY>
+class Sequencer {
```

Comment:
maybe put this class in `util/sequencer.h`, so it can be used by other RPC clients as well.

Reply:
fixed